### PR TITLE
Add PUT and DELETE

### PR DESCRIPTION
**Pull request by conghuat**
change the PUT request

### CODE REVIEW

File: rpi_api/api.py [hunk 1]
```diff
@@ -84,9 +78,22 @@ def post(self, command: str) -> dict:
...
+    def delete(self) -> dict:
+        """DELETE command to delete all LED"""
+        VALID_LEDS.clear()
+        leds.clear()
+        return {
+            "pins": [get_led(led_pin) for led_pin in leds]
+        }
+    
+    
+class LED(Resource): # signle assess to the LED
```
Comment (BTx123) [inline rpi_api/api.py]:
```suggestion
class LED(Resource):
```

File: rpi_api/api.py [hunk 2]
```diff
@@ -66,7 +60,7 @@ def get(self) -> dict:
             "pins": [get_led(led_pin) for led_pin in leds]
         }
 
-    def post(self, command: str) -> dict:
+    def post(self, command: str) -> dict: # makes it on or off
```
Comment (BTx123) [inline rpi_api/api.py]:
```suggestion
    def post(self, command: str) -> dict:
```

File: rpi_api/api.py [hunk 3]
```diff
@@ -47,15 +41,15 @@ def get_led(led_pin: int) -> dict:
...
+class HelloWorld(Resource): # test the thing is working
     """Demo 'Hello World' resource."""
 
     def get(self):
         LOGGER.debug("GET: Hello, World!")
         return {"hello": "world"}
 
 
-class LEDS(Resource):
+class LEDS(Resource): # give the multiple LEDs
```
Comment (BTx123) [inline rpi_api/api.py]:
```suggestion
class LEDS(Resource):
```

File: rpi_api/api.py [hunk 4]
```diff
@@ -109,9 +115,29 @@ def post(self, led_pin: int, command: str) -> dict:
         elif command == "toggle":
             leds[led_pin].toggle()
         else:
-            abort(404, message="Command not available.")
+            abort(404, message="Command not available.") # tell us that the event is not valid
```
Comment (BTx123) [inline rpi_api/api.py]:
```suggestion
            abort(404, message="Command not available.")
```

File: rpi_api/api.py [hunk 5]
```diff
@@ -22,23 +15,24 @@
...
 DEBUG = True
 
-# Routing API
+# Routing API   host/api/led/17
 ROUTE_API = "/api"
 ROUTE_LEDS = "/leds"
 
 # LED numbers should refer to GPIO number not RPi pin number
-VALID_LEDS = [17, 27, 22]
+VALID_LEDS = [17, 27, 22] # the ID  for the LED
```
[the remark this answers is not in the document]
Comment (BTx123) [reply rpi_api/api.py]: Remove this line, use leds dictionary to check for pin validity.

File: rpi_api/api.py [hunk 6]
```diff
@@ -109,9 +115,29 @@ def post(self, led_pin: int, command: str) -> dict:
...
         else:
-            abort(404, message="Command not available.")
+            abort(404, message="Command not available.") # tell us that the event is not valid
 
         return get_led(led_pin)
+    
+    def put(self, led_pin: int) -> dict:
+        """PUT command to create new LED"""
+        VALID_LEDS.append(led_pin)
+        leds[led_pin] = gpiozero.LED(led_pin)
```
Comment (conghuat) [inline rpi_api/api.py]: Give a message tells the user that this led pin number is already in the data base

**Pull request merged**